src/main/groovy/com/google/gcloud/GCloudPlugin.groovy

```diff
@@ -1,33 +1,131 @@
+                        deployTask.setDescriptors(configuration.descriptors)
+                        deployTask.description = 'Deploy the ' + configuration.name + ' configuration to Google Cloud';
+                        deployTask.group = TASK_GROUP_GCLOUD;
+                        configuration.modules.each { Project subProject ->
+                            addStageTaskDependency(deployTask, subProject)
+                        }
+                    }
+                })
+
+                // only allow local run/stop for configurations with app.yaml
+                if (configuration.getDescriptors().contains("app.yaml")) {
+                    // run task
+                    String runTaskName = TASK_GCLOUD_RUN + configuration.name.capitalize()
+                    project.tasks.create(runTaskName, RunTask, new Action<RunTask>() {
+                        @Override
+                        void execute(RunTask runTask) {
+                            runTask.setArgs(configuration.args + configuration.runArgs)
+                            runTask.setModules(configuration.modules)
+                            runTask.setDescriptors(configuration.APP_DESCRIPTOR)
+                            runTask.setAsync(configuration.runAsync)
+                            runTask.description = 'Run the ' + configuration.name + ' configuration on Dev App Server';
+                            runTask.group = TASK_GROUP_GCLOUD;
+                            configuration.modules.each { Project subProject ->
+                                addStageTaskDependency(runTask, subProject)
+                            }
+                        }
+                    })
+
+                    // stop task
+                    String stopTaskName = TASK_GCLOUD_STOP + configuration.name.capitalize()
+                    project.tasks.create(stopTaskName, StopTask, new Action<StopTask>() {
+                        @Override
+                        void execute(StopTask stopTask) {
+                            stopTask.description = 'Stop the ' + configuration.name + ' configuration on Dev App Server';
+                            stopTask.group = TASK_GROUP_GCLOUD;
+                            def adminHost = configuration.args.get("admin-host") ?: configuration.args.get("--admin-host")
+                            if (adminHost) {
+                                stopTask.setAdminHost(adminHost.toString())
+                            }
+                        }
+                    })
+                }
+            }
+        }
+    }
 
-        project.tasks.create(TASK_GCLOUD_APP_RUN, GCloudAppRunTask)
-        project.tasks.create(TASK_GCLOUD_APP_DEPLOY, GCloudAppDeployTask)
+    // figure out if the subProject has been evaluated or not before trying to add task dependency
+    private static void addStageTaskDependency(Task task, Project subProject) {
+        if (subProject.state.executed) {
```

Comment:
The purpose of this section is to ensure that the "appengineStage" task exists.  The order of evaluation (where the build file is parsed/executed - https://docs.gradle.org/current/userguide/build_lifecycle.html) allows for some project's configuration to be unknown at this point. If the subproject hasn't been evaluated yet, we can just wait to add the task dependnecy after it has been evaluated.  If they have been evaluated already at this point (subProject.state.executed = true) then the task is know and can be executed. Unfortunately we can't just put everything in the "afterEvaluate", as for already evaluated projects this results in a no-op.